sphinx_ext/deephaven_autodoc.py

```diff
@@ -0,0 +1,311 @@
+from __future__ import annotations
+
+import re
+import json
+from typing import TypedDict, Union, List, Dict
+
+import docutils.nodes
+import sphinx.addnodes
+from sphinx.ext.autodoc.directive import AutodocDirective
+
+from sphinx.application import Sphinx
+
+
+class ParamData(TypedDict, total=False):
+    name: str
+    type: str
+    description: str
+    # default is not required
+    default: str | None
```

Comment:
Can't `default` already be omitted because `total=False`? 

Actually I see what's up here. You want `name`, `type`, `description` required, but not `default`. I think the correct way to type this is to use [`typing.NotRequired`](https://docs.python.org/3/library/typing.html#typing.NotRequired)

```suggestion
class ParamData(TypedDict):
    name: str
    type: str
    description: str
    default: NotRequired[str]
```